merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/FixedRateListener.java

```diff
@@ -0,0 +1,46 @@
+package gov.nasa.jpl.aerie.merlin.server.services;
+
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+public interface FixedRateListener<T> extends AutoCloseable {
+  void updateValue(T t);
+
+  @Override
+  void close();
+
+  /**
+   * Call reporter.accept at fixed intervals using the given period. The value passed to the consumer is the
+   * most recent argument to `updateValue`.
+   *
+   * @param reporter The consumer to call periodically
+   * @param initialValue The initial value to pass to the given consumer
+   * @param pollPeriodMillis The interval at which to call the given consumer
+   * @return a consumer with an additional close method
+   */
+  static <T> FixedRateListener<T> callAtFixedRate(final Consumer<T> reporter, final T initialValue, final long pollPeriodMillis) {
```

Comment:
Very minor suggestion, can be ignored aside from `pollPeriodMillis`, which needs to be renamed and the `@return`, which seems incorrect since the consumer is inaccessible in the `FixedRateListener` object returned.

```suggestion
  /**
   * Call reporter.accept at fixed intervals using the given period. The value passed to reporter.accept is the
   * most recent argument to `updateValue`.
   *
   * @param reporter The Consumer to call periodically
   * @param initialValue The initial value to pass to reporter
   * @param reportPeriodMillis The interval at which to call reporter.accept
   * @return a FixedRateListener with an additional close method
   */
  static <T> FixedIntervalListener<T> callAtFixedInterval(final Consumer<T> reporter, final T initialValue, final long reportPeriodMillis) {
```